File: Python/SQLAlchemy/src/models/person.py
import uuid

from sqlalchemy import Column, String
from sqlalchemy.dialects.sqlite import DATE
from sqlalchemy.orm import relationship

from .base import Base


class Person(Base):
    __tablename__ = 'people'

    _id = Column(
        String,
        name='id',
        primary_key=True,
        default=lambda: str(uuid.uuid4())
    )

    name = Column(String)
    address = Column(String)
    birth_date = Column(DATE)
    _employees = relationship('Employee', cascade="all, delete")

    @property
    def id(self):
        return self._id

    def __str__(self):
        return (
            "Person { "
            f"Id: {self.id}, "
            f"Name: {self.name}, "
            f"Address: {self.address}, "
            f"BirthDate: {self.birth_date.strftime('%d.%m.%Y')} "
            "}"
        )

File: TypeScript/MicroORM/models/Person.ts
import { Entity, Property, PrimaryKey, DateType } from '@mikro-orm/core';
import { v4 } from 'uuid';

@Entity({ tableName: 'people' })
export class Person {
    @PrimaryKey()
    public id = v4();

    @Property()
    public name!: string;

    @Property()
    public address!: string;

    @Property({ type: DateType })
    public birthDate!: Date;

    string(): string {
        return `{ Person: { Id: ${this.id}, Name: ${this.name}, Address: ${this.address}, BirthDate: ${this.birthDate.toISOString().split('T')[0]} } }`;
    }
}
